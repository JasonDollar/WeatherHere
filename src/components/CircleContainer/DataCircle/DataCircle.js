import React from 'react'
import PropTypes from 'prop-types'

import Icon from '../../Icon/Icon'

import classes from './DataCircle.module.scss'

const DataCircle = ({
  type, data, unit, additionalData, text,
}) => {
  const graphColorPrimary = getComputedStyle(document.documentElement).getPropertyValue('--color-primary')
  const graphColorSecondary = getComputedStyle(document.documentElement).getPropertyValue('--color-primary-offset')
  // const colorGrey = getComputedStyle(document.documentElement).getPropertyValue('--color-back')
  const colorGrey = getComputedStyle(document.documentElement).getPropertyValue('--color-grey-4')
  const style = { transform: `rotate(${additionalData - 90}deg)` }
  let conic 
  let rotateCSSVarRight
  let rotateCSSVarLeft
  let colorCSSRight
  let colorCSSLeft
  let elementName
  // determine which css variables this el. will be using
  if (type === text.humidity) {
    
    rotateCSSVarRight = '--s-right-rotate1'
    rotateCSSVarLeft = '--s-left-rotate1'
    colorCSSRight = '--s-right-color1'
    colorCSSLeft = '--s-left-color1'
    elementName = 'humidity'
  } else if (type === text.precipProb) {
    
    rotateCSSVarRight = '--s-right-rotate2'
    rotateCSSVarLeft = '--s-left-rotate2'
    colorCSSRight = '--s-right-color2'
    colorCSSLeft = '--s-left-color2'
    elementName = 'precipProb'
  } else if (type === text.cloudCov) {
    rotateCSSVarRight = '--s-right-rotate3'
    rotateCSSVarLeft = '--s-left-rotate3'
    colorCSSRight = '--s-right-color3'
    colorCSSLeft = '--s-left-color3'
    elementName = 'cloudCov'
  }

  // change of css variables to imitate conic-gradient
  if (data <= 0) {
    document.documentElement.style.setProperty(colorCSSRight, colorGrey)
    document.documentElement.style.setProperty(rotateCSSVarRight, '0deg')
    document.documentElement.style.setProperty(colorCSSLeft, colorGrey)
    document.documentElement.style.setProperty(rotateCSSVarLeft, '0deg')
    conic = { background: `conic-gradient(${colorGrey} 0% 100%)` }
  } else if (data < 50) {
    document.documentElement.style.setProperty(colorCSSRight, colorGrey)
    document.documentElement.style.setProperty(rotateCSSVarRight, `${parseInt(data * 3.6)}deg`)
    document.documentElement.style.setProperty(colorCSSLeft, colorGrey)
    document.documentElement.style.setProperty(rotateCSSVarLeft, '0deg')
    conic = { background: `conic-gradient(${graphColorPrimary} 0%, ${graphColorSecondary} ${data}%, ${colorGrey} ${data}% 100%)` }
  } else if (data === 50) {
    document.documentElement.style.setProperty(rotateCSSVarRight, '0deg')
    document.documentElement.style.setProperty(colorCSSRight, graphColorSecondary)
    document.documentElement.style.setProperty(rotateCSSVarLeft, '0deg')
    document.documentElement.style.setProperty(colorCSSLeft, colorGrey)
    conic = { background: `conic-gradient(${graphColorPrimary} 0%, ${graphColorSecondary} ${data}%, ${colorGrey} ${data}% 100%)` }
  } else if (data > 50) {
    document.documentElement.style.setProperty(rotateCSSVarRight, '0deg')
    document.documentElement.style.setProperty(colorCSSRight, graphColorSecondary)
    document.documentElement.style.setProperty(rotateCSSVarLeft, `${parseInt(data * 3.6) - 180}deg`)
    document.documentElement.style.setProperty(colorCSSLeft, colorGrey)
    conic = { background: `conic-gradient(${graphColorSecondary} 0%, ${graphColorSecondary} 50%,${graphColorPrimary} ${data}%, ${graphColorSecondary} ${data}%, ${colorGrey} ${data}% 100%)` }
  } else if (data >= 100) {
    document.documentElement.style.setProperty(rotateCSSVarRight, '0deg')
    document.documentElement.style.setProperty(colorCSSRight, graphColorSecondary)
    document.documentElement.style.setProperty(rotateCSSVarLeft, '0deg')
    document.documentElement.style.setProperty(colorCSSLeft, graphColorSecondary)
    conic = { background: `conic-gradient(${graphColorSecondary} 0%, ${graphColorSecondary} 50%,${graphColorPrimary} ${data}%, ${graphColorSecondary} ${data}%, ${colorGrey} ${data}% 100%)` }
  }
 
  return (
    <div className={`${classes.DataCircle} ${classes[elementName]}`}>
      <div className={classes.dataColor} style={conic} />
      <div className={classes.dataFront} />
      <div className={classes.dataContainer}>
        {additionalData !== null ? (
          <div style={style} className={classes.bigArrow}>
            {additionalData !== null ? <Icon icon="arrow" viewBox="0 0 24 24" /> : null}
          </div>  
        ) : null }
        <span>
          {data + unit}
        </span> 
        
        <span>
          {type}
        </span>
      </div>
      
    </div>
  ) 
}

export default DataCircle


DataCircle.propTypes = {
  type: PropTypes.string.isRequired,
  data: PropTypes.number.isRequired,
  unit: PropTypes.string.isRequired, 
  additionalData: PropTypes.number,
  text: PropTypes.objectOf(PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.arrayOf(PropTypes.string),
  ])).isRequired,
}

DataCircle.defaultProps = {
  additionalData: null,
}

