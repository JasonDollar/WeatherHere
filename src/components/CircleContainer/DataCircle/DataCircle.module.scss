.DataCircle {

  position: relative;
  height: var(--circle-dimension-small);
  width: var(--circle-dimension-small);
  border: 1px solid #ccc;
  border-radius: 50%;
  margin: 1rem auto;

  @media only screen and (min-width: 576px) {
    height: var(--circle-dimension);
    width: var(--circle-dimension);
    font-size: var(--normal-text-font-size);
  }

  & .dataContainer {

    font-size: var(--normal-text-font-size);
    position: absolute;
    top: 47%;
    left: 50%;
    transform: translate(-50%, -50%);
    display: flex;
    flex-direction: column;
    align-items: center;
    text-align: center;
    z-index: 5;

    @media only screen and (min-width: 576px) {

      font-size: var(--big-text-font-size);
    }

    & span {
      display: block;
    }

    & .bigArrow {
      margin-bottom: 1rem;
    }

    & svg {

      height: var(--head-font-size);
      width: var(--head-font-size);
      transform: scale(1.5);
      fill: #1e3a6f;
    }
  }

  & .dataFront {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    z-index: 1;
    width: 12rem;
    height: 12rem;
    background-color: var(--color-back);
    border: 1px solid transparent;
    border-radius: 50%;

    @media only screen and (min-width: 576px) {
      width: 16rem;
      height: 16rem;
    }
  }

}

// .arrow {
//   width: 100%;
//   height: 100%;
//   display: block;
//   position: absolute;
//   top: 0;

//   &:after {
//     content: "";
//     width: 0;
//     height: 0;
//     border-left: 5px solid transparent;
//     border-right: 5px solid transparent;
//     border-bottom: 10px solid red;
//     position: absolute;
//     top: -6px;
//     left: 50%;
//     margin-left: -5px;
//     z-index: 99;
//   }
// }