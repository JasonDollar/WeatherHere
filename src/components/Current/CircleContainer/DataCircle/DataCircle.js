import React from 'react'
import PropTypes from 'prop-types'
// import {
//   PieChart, Pie, Sector, Cell, ResponsiveContainer, 
// } from 'recharts'
import Icon from '../../../Icon/Icon'

import classes from './DataCircle.module.scss'

const DataCircle = ({
  type, data, unit, additionalData, graphColor,
}) => {
  const style = { transform: `rotate(${additionalData - 90}deg)` }
  const conic = { background: `conic-gradient(${graphColor.primary}99 0 ${data}%, #fff ${data}% 100%)` }

  return (
    <div className={classes.DataCircle} style={conic}>
      
      <div className={classes.dataFront} />
      <div className={classes.dataContainer}>
        {additionalData !== null ? (
          <div style={style} className={classes.bigArrow}>
            {additionalData !== null ? <Icon icon="arrow" viewBox="0 0 24 24" /> : null}
          </div>  
        ) : null }
        <span>
          {data + unit}
        </span> 
        
        <span>
          {type}
        </span>
      </div>
      
      {/*additionalData !== null ? <div className={classes.arrow} style={style2} /> : null*/}
    </div>
  ) 
  // return (
  //   <ResponsiveContainer heigh="100%" width="100%">
  //     <PieChart height={300} width={300}>
  //       <Pie
  //         data={wholeData}
  //         cx="50%"
  //         cy="50%"
  //         outerRadius={80} 
  //         fill="#8884d8"
  //       />
        
  //     </PieChart>
  //   </ResponsiveContainer>
  // )
}

export default DataCircle


DataCircle.propTypes = {
  type: PropTypes.string.isRequired,
  data: PropTypes.number.isRequired,
  unit: PropTypes.string.isRequired, 
  additionalData: PropTypes.number,
}