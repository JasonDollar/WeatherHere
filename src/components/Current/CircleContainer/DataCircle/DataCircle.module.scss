.DataCircle {

  position: relative;
  height: 20rem;
  width: 20rem;
  border: 1px solid #ccc;
  border-radius: 50%;
  margin: 1rem auto;

  & .dataContainer {
    font-size: var(--big-text-font-size);
    position: absolute;
    top: 47%;
    left: 50%;
    transform: translate(-50%, -50%);
    display: flex;
    flex-direction: column;
    align-items: center;
    text-align: center;
    z-index: 5;


    & span {
      display: block;
    }

    & .bigArrow {
      margin-bottom: 1rem;
    }

    & svg {

      height: var(--head-font-size);
      width: var(--head-font-size);
      transform: scale(1.5);
      fill: #1e3a6f;
    }
  }

  & .dataFront {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    z-index: 1;
    width: 16rem;
    height: 16rem;
    background-color: #fff;
    border: 1px solid transparent;
    border-radius: 50%;
  }

}

// .arrow {
//   width: 100%;
//   height: 100%;
//   display: block;
//   position: absolute;
//   top: 0;

//   &:after {
//     content: "";
//     width: 0;
//     height: 0;
//     border-left: 5px solid transparent;
//     border-right: 5px solid transparent;
//     border-bottom: 10px solid red;
//     position: absolute;
//     top: -6px;
//     left: 50%;
//     margin-left: -5px;
//     z-index: 99;
//   }
// }