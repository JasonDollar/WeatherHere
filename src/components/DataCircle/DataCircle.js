import React from 'react'
// import { Doughnut } from 'react-chartjs-2'

import classes from './DataCircle.module.scss'

const DataCircle = ({
  type, data, unit, additionalData, 
}) => {
  const style = { transform: `rotate(${additionalData}deg)` }
  const conic = { background: `conic-gradient(rgba(198,210,60,.5) 0 ${data}%, #fff ${data}% 100%)` }
  return (
    <div className={classes.DataCircle} style={conic}>
      
      <div className={classes.dataFront} />
      <div className={classes.dataContainer}>
        <span>
          {data + unit}
        </span>
        
        <span>
          {type}
        </span>
      </div>
      
      {additionalData !== null ? <div className={classes.arrow} style={style} /> : null}
    </div>
  )
}

export default DataCircle
