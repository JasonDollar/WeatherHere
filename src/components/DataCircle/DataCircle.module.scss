.DataCircle {
  
  position: relative;
  height: 20rem;
  width: 20rem;
  border: 1px solid goldenrod;
  border-radius: 50%;

  .dataContainer {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    display: flex;
    flex-direction: column;
    align-items: center;
    text-align: center;
  }

  span {
    display: block;
  }
}

.arrow {
  width: 100%;
  height: 100%;
  display: block;
  position: absolute;
  top: 0;
  
  &:after {
    content: "";
    width: 0;
    height: 0;
    border-left: 5px solid transparent;
    border-right: 5px solid transparent;
    border-bottom: 10px solid red;
    position: absolute;
    top: -6px;
    left: 50%;
    margin-left: -5px;
    z-index: 99;
  }
}