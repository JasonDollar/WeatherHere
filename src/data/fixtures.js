import weatherData from './random'
import languageText from './text-locale' 

export const appTestData = {
  weather: weatherData,
  units: 'si',
  language: 'pl',
  text: languageText,
  timezone: 'Europe/Warsaw',
  locationShortName: 'Plock',
  
}