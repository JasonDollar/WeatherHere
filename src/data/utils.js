import moment from 'moment-timezone'

const getTimeFromSeconds = (timestamp, timezone) => {
  const timestampInMiliSec = timestamp * 1000
  if (timezone === null) {
    return {
      month: moment(timestampInMiliSec).format('D.M'),
      // day === week day
      day: moment(timestampInMiliSec).format('e'),
      
      hour: moment.tz(timestampInMiliSec, timezone).format('HH:mm'),
      
    } 
  }
  return {
    month: moment.tz(timestampInMiliSec, timezone).format('D.M'),
    // day === week day
    day: moment.tz(timestampInMiliSec, timezone).format('d'),
    hour: moment.tz(timestampInMiliSec, timezone).format('HH:mm'),
  } 
}

const formatNumber = number => Math.floor(number * 10) / 10


export { getTimeFromSeconds, formatNumber }