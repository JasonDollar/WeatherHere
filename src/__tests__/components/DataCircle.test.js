import React from 'react'
import { shallow } from 'enzyme'
import { appTestData as data } from '../../data/fixtures'
import DataCircle from '../../components/CircleContainer/DataCircle/DataCircle' 
import { formatNumber } from '../../data/utils'

let typeHumidity = data.text.pl.current.humidity
let typeWind = data.text.pl.current.wind
let dataHumidity = formatNumber(data.weather.currently.humidity * 100)
let dataWind = formatNumber(data.weather.currently.windSpeed)
let unitPercent = '%'
let unitWind = 'm/s'
let additionalData = data.weather.currently.windBearing
let text = data.text.pl.current
let wrapper



describe('DataCircle test', () => {
  describe('DataCircle comp. with humidity data (works for most types)', () => {
    beforeAll(() => {

      global.document.documentElement.style.setProperty = jest.fn()
      wrapper = shallow(
        <DataCircle 
          type={typeHumidity}
          data={dataHumidity}
          unit={unitPercent}
          text={text}
        />,
      )
    })
    
    test('Data Circle humidity snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })

    test('rendered proper data', () => {
      const element = wrapper.find('.dataContainer span').at(0)
      expect(element.render().text()).toBe(`${dataHumidity}${unitPercent}`)
    })
    test('rendered proper data type', () => {
      const element = wrapper.find('.dataContainer span').at(1)
      expect(element.render().text()).toBe(typeHumidity)
    })
  })
  describe('DataCircle comp. with wind data', () => {
    beforeAll(() => {
      wrapper = shallow(
        <DataCircle 
          type={typeWind}
          data={dataWind}
          unit={unitWind}
          additionalData={additionalData}
          text={text}
        />,
      )
    })
    
    test('Data Circle wind snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })

    test('rendered proper data', () => {
      const element = wrapper.find('.dataContainer span').at(0)
      expect(element.render().text()).toBe(`${dataWind}${unitWind}`)
    })
    test('rendered proper data type', () => {
      const element = wrapper.find('.dataContainer span').at(1)
      expect(element.render().text()).toBe(typeWind)
    })
    test('additionaldata should be used to apply rotate to arrow element', () => {
      const element = wrapper.find('.bigArrow')
      expect(element.prop('style')).toEqual({ transform: `rotate(${additionalData - 90}deg)` })
    })
  })
})

/*
21: "--s-right-rotate1"
​
22: "--s-right-color1"
​
23: "--s-left-rotate1"
​
24: "--s-left-color1"
​
25: "--s-right-color2"
​
26: "--s-right-rotate2"
​
27: "--s-left-color2"
​
28: "--s-left-rotate2"
​
29: "--s-right-color3"
​
30: "--s-right-rotate3"
​
31: "--s-left-color3"
​
32: "--s-left-rotate3"
*/