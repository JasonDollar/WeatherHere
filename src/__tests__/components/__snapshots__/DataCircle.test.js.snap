// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataCircle test DataCircle comp. with humidity data (works for most types) Data Circle humidity snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DataCircle
    additionalData={null}
    data={94}
    text={
      Object {
        "appTemp": "Temperatura odczuwalna",
        "cloudCov": "Zachmurzenie",
        "date": "Data",
        "dewPoint": "Punkt rosy",
        "humidity": "Wilgotność",
        "icon": "Ikona pogody",
        "location": "Nieznana lokacja",
        "moon": Array [
          "Nów",
          "Wschodzący sierp",
          "Pierwsza kwadra",
          "Wschodzący księżyc",
          "Pełnia",
          "Ubywający księżyc",
          "Trzecia kwadra",
          "Ubywający sierp",
        ],
        "moonphase": "Faza księżyca",
        "ozone": "Ozon",
        "precipInt": "Intensywność opadu",
        "precipProb": "Opady",
        "precipType": "Rodzaj opadu",
        "pressure": "Ciśnienie",
        "sun": "Słońce",
        "sunrise": "Wschód słońca",
        "sunset": "Zachód słońca",
        "today": "Dziś",
        "uvIndex": "Indeks UV",
        "visibility": "Widoczność",
        "wind": "Wiatr",
        "windGust": "Wiatr w porywach",
      }
    }
    type="Wilgotność"
    unit="%"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="dataColor"
          style={
            Object {
              "background": "conic-gradient( 0%,  50%, 94%,  94%,  94% 100%)",
            }
          }
        />,
        <div
          className="dataFront"
        />,
        <div
          className="dataContainer"
        >
          <span>
            94%
          </span>
          <span>
            Wilgotność
          </span>
        </div>,
      ],
      "className": "DataCircle humidity",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "dataColor",
          "style": Object {
            "background": "conic-gradient( 0%,  50%, 94%,  94%,  94% 100%)",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "dataFront",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            null,
            <span>
              94%
            </span>,
            <span>
              Wilgotność
            </span>,
          ],
          "className": "dataContainer",
        },
        "ref": null,
        "rendered": Array [
          null,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "94%",
            },
            "ref": null,
            "rendered": "94%",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Wilgotność",
            },
            "ref": null,
            "rendered": "Wilgotność",
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="dataColor"
            style={
              Object {
                "background": "conic-gradient( 0%,  50%, 94%,  94%,  94% 100%)",
              }
            }
          />,
          <div
            className="dataFront"
          />,
          <div
            className="dataContainer"
          >
            <span>
              94%
            </span>
            <span>
              Wilgotność
            </span>
          </div>,
        ],
        "className": "DataCircle humidity",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "dataColor",
            "style": Object {
              "background": "conic-gradient( 0%,  50%, 94%,  94%,  94% 100%)",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "dataFront",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              null,
              <span>
                94%
              </span>,
              <span>
                Wilgotność
              </span>,
            ],
            "className": "dataContainer",
          },
          "ref": null,
          "rendered": Array [
            null,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "94%",
              },
              "ref": null,
              "rendered": "94%",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Wilgotność",
              },
              "ref": null,
              "rendered": "Wilgotność",
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`DataCircle test DataCircle comp. with wind data Data Circle wind snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DataCircle
    additionalData={99}
    data={5.7}
    text={
      Object {
        "appTemp": "Temperatura odczuwalna",
        "cloudCov": "Zachmurzenie",
        "date": "Data",
        "dewPoint": "Punkt rosy",
        "humidity": "Wilgotność",
        "icon": "Ikona pogody",
        "location": "Nieznana lokacja",
        "moon": Array [
          "Nów",
          "Wschodzący sierp",
          "Pierwsza kwadra",
          "Wschodzący księżyc",
          "Pełnia",
          "Ubywający księżyc",
          "Trzecia kwadra",
          "Ubywający sierp",
        ],
        "moonphase": "Faza księżyca",
        "ozone": "Ozon",
        "precipInt": "Intensywność opadu",
        "precipProb": "Opady",
        "precipType": "Rodzaj opadu",
        "pressure": "Ciśnienie",
        "sun": "Słońce",
        "sunrise": "Wschód słońca",
        "sunset": "Zachód słońca",
        "today": "Dziś",
        "uvIndex": "Indeks UV",
        "visibility": "Widoczność",
        "wind": "Wiatr",
        "windGust": "Wiatr w porywach",
      }
    }
    type="Wiatr"
    unit="m/s"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="dataColor"
          style={
            Object {
              "background": "conic-gradient( 0%,  5.7%,  5.7% 100%)",
            }
          }
        />,
        <div
          className="dataFront"
        />,
        <div
          className="dataContainer"
        >
          <div
            className="bigArrow"
            style={
              Object {
                "transform": "rotate(9deg)",
              }
            }
          >
            <Icon
              icon="arrow"
              title="icon"
              viewBox="0 0 24 24"
            />
          </div>
          <span>
            5.7m/s
          </span>
          <span>
            Wiatr
          </span>
        </div>,
      ],
      "className": "DataCircle undefined",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "dataColor",
          "style": Object {
            "background": "conic-gradient( 0%,  5.7%,  5.7% 100%)",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "dataFront",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="bigArrow"
              style={
                Object {
                  "transform": "rotate(9deg)",
                }
              }
            >
              <Icon
                icon="arrow"
                title="icon"
                viewBox="0 0 24 24"
              />
            </div>,
            <span>
              5.7m/s
            </span>,
            <span>
              Wiatr
            </span>,
          ],
          "className": "dataContainer",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Icon
                icon="arrow"
                title="icon"
                viewBox="0 0 24 24"
              />,
              "className": "bigArrow",
              "style": Object {
                "transform": "rotate(9deg)",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "icon": "arrow",
                "title": "icon",
                "viewBox": "0 0 24 24",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "5.7m/s",
            },
            "ref": null,
            "rendered": "5.7m/s",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Wiatr",
            },
            "ref": null,
            "rendered": "Wiatr",
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="dataColor"
            style={
              Object {
                "background": "conic-gradient( 0%,  5.7%,  5.7% 100%)",
              }
            }
          />,
          <div
            className="dataFront"
          />,
          <div
            className="dataContainer"
          >
            <div
              className="bigArrow"
              style={
                Object {
                  "transform": "rotate(9deg)",
                }
              }
            >
              <Icon
                icon="arrow"
                title="icon"
                viewBox="0 0 24 24"
              />
            </div>
            <span>
              5.7m/s
            </span>
            <span>
              Wiatr
            </span>
          </div>,
        ],
        "className": "DataCircle undefined",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "dataColor",
            "style": Object {
              "background": "conic-gradient( 0%,  5.7%,  5.7% 100%)",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "dataFront",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="bigArrow"
                style={
                  Object {
                    "transform": "rotate(9deg)",
                  }
                }
              >
                <Icon
                  icon="arrow"
                  title="icon"
                  viewBox="0 0 24 24"
                />
              </div>,
              <span>
                5.7m/s
              </span>,
              <span>
                Wiatr
              </span>,
            ],
            "className": "dataContainer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Icon
                  icon="arrow"
                  title="icon"
                  viewBox="0 0 24 24"
                />,
                "className": "bigArrow",
                "style": Object {
                  "transform": "rotate(9deg)",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "icon": "arrow",
                  "title": "icon",
                  "viewBox": "0 0 24 24",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "5.7m/s",
              },
              "ref": null,
              "rendered": "5.7m/s",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Wiatr",
              },
              "ref": null,
              "rendered": "Wiatr",
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
